eng/Directory.Packages.props

```diff
@@ -194,18 +194,15 @@
     <PackageVersion Include="System.IO.Hashing" Version="$(SystemIOHashingVersion)" />
     <PackageVersion Include="System.IO.Pipelines" Version="$(SystemIOPipelinesVersion)" />
     <PackageVersion Include="System.IO.Pipes.AccessControl" Version="5.0.0" />
-    <PackageVersion Include="System.Memory" Version="4.5.5" />
-    <PackageVersion Include="System.Numerics.Vectors" Version="4.5.0" />
-    <PackageVersion Include="System.Runtime.CompilerServices.Unsafe" Version="6.0.0" />
-    <PackageVersion Include="System.Runtime.Loader" Version="4.3.0" />
+    <PackageVersion Include="System.Memory" Version="4.6.0" />
```

Comment:
> can someone on your team please take a look at the `MissingMethodException` failure. This looks related to the `System.Memory` version bump.

I think the MissingMethodException happens because MSBuild loads our custom build task (SemanticSearch.BuildTask.dll) which now depends on System.Memory 4.6.0, but MSBuild itself depends on an older version of System.Memory, so there's a mismatch.

As is linked in the other PR, the solution might be https://github.com/dotnet/msbuild/pull/11038.

Reply:
Yes that's my standing theory but we should still temporarily be able to consume a newer version by adding the assembly app-local. It seems like this isn't happening here.

Reply:
How would app local help here though? The version in msbuild will be loaded first, it will win and the one local to the task will not be considered.